### ISSUE TITLE
Aliasing Out-File is dangerous

### ISSUE THREAD

**Issue by vors** [weswigham/GithubProvider#6]
`out-file` is a built-in cmdlet. Aliasing it can break somebodies existing scripts, if it's not 100% compatible. This happened with me. When I dig into it, it turns out I have GithubFS module imported and it altered my aliases.

Here is an exception that I get.

```
> '1' | out-file -Encoding ASCII foo
Get-Item : Cannot bind argument to parameter 'Path' because it is an empty string.
At C:\Program Files\WindowsPowerShell\Modules\GithubFS\1.0.1\GithubFS.psm1:44 char:19
+ ...  $(get-item $FilePath.Substring(0,$FilePath.LastIndexOf(':')+1)).PSPr ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Item], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.GetI
   temCommand
```

Also, even if it's implemented properly such aliasing may add performance burn that is hard to track.

**Comment by weswigham** [comment]
On the other hand, AFAIK aliasing `out-file` is the only method by which you can use a non-fs provider with the output redirection operator.

**Comment by weswigham** [comment]
BTW, The `Out-Github` function which aliases over `Out-File` cmdlet is the modified output of the canonical proxy generating cmdlets:

``` powershell
$metadata = New-Object System.Management.Automation.CommandMetaData (Get-Command Out-File)
$src = [System.Management.Automation.ProxyCommand]::Create($metadata)
```

**Comment by vors** [comment]
oh, that's interesting. Looks like a flow in `ProxyCommand`.

**Comment by weswigham** [comment]
#7 should fix this. I originally didn't think `Get-Item` could (validly) return a null value.

**Comment by weswigham** [comment]
Though I'll probably hold off merging it until I get around to fixing CI.

**Comment by vors** [comment]
UPD: I tried generate `Out-File2` from `$src` and it works fine.
I thought I can repro the binding Path problem generated from metadata, but I acutally used **your** aliased verison when I did it :)

**Comment by weswigham** [comment]
Yeah - it looks like it choked on nonexistant paths. I'm working on a better way of getting the provider for a given path than attempting get-item, though.

**Comment by weswigham** [comment]
The issues you were having _should_ be resolved. Not only is my path handling more robust (ie, I don't touch it), I also fixed a few minor issues I discovered which were preventing tests from passing (for example, `gc` and `sc` on github paths don't support flags like encoding, etc - however, generic tools like `pester` expect those flags to be accepted. Now the provider internally accepts those flags (even if it doesn't yet use them).
